### PR TITLE
12 database setup

### LINKED ISSUE #12: Database Setup
setup database 

### Acceptance Criteria:

- [x] create database files 
- [x] create table files 
- [x] create connection file 
- [x] create bulid file
- [x] create test files 
- [x] add dummy data

### Database scema:
unknown (10)

### PR DESCRIPTION
**Pull request by AlaaShurrab**
#### fixes #12 
=======

creating the database tables, file structure and connect and build files 
---

#### for  test 
you should have a URL for the database in the .env file with the name: ``` DATABASE_URL```

#### You can use
 
use the command ```yarn test ``` to check the utiles functions 
use command ``` yarn build:db``` to build the local database tables 
use query
```sql
UPDATE patients
SET first_name = 'any first name'
where id = 1 RETURNING * ;
```
to test the trigger and check the the update_at 

---

#### for team
you should check the comments for the getClinet function on the connect file in the database 

---

#### Quick Pull Request Checklist

Please make sure you ensure the following has been done before marking your PR as awaiting review. For a full checklist and process review
- [x] Documentation added
- [x] Tests
- [x]  Add dummy data
- [x] Ready to be merged

### CODE REVIEW

File: server/src/database/models/patient-general-therapy/schema.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+DROP TABLE IF EXISTS "patient_general_therapy" CASCADE;
+
+CREATE TABLE "patient_general_therapy" (
+  "id" SERIAL PRIMARY KEY,
+  "patient_id" INTEGER REFERENCES patients(id) NOT NULL,
+  "global_therapy_id" INTEGER REFERENCES global_therapy(id) NOT NULL,
```
Comment (Osama-you) [inline server/src/database/models/patient-general-therapy/schema.sql]:
```suggestion
  "general_therapy_id" INTEGER REFERENCES general_therapy(id) NOT NULL,
```
Looks like you forgot to edit the name here.

File: server/src/database/dummy-data/patient-general-therapy.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+const createTherapy = async ({
+  patientId,
+  generalTherapyId,
+  therapyNumber,
+  notes,
+}) => {
+  const sql = `INSERT INTO patient_general_therapy(
+    patient_id,
+    global_therapy_id,
+    therapy_number,
```
Comment (Osama-you) [inline server/src/database/dummy-data/patient-general-therapy.js]:
```suggestion
    patient_id,
   general_therapy_id,
    therapy_number,
```
Also, fix it in the dummy data.

File: server/src/database/dummy-data/index.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  createdData.users = await createUsers(createdData);
+  createdData.patients = await createPatients(createdData);
+  createdData.generalTherapy = await createGeneralTherapy(createdData);
+  createdData.teethTherapy = await createTeethTherapy(createdData);
+  createdData.patientTeethTherapy = await createPatientTeethTherapy(
+    createdData
+  );
+  //   createdData.patientGeneralTherapy = await createPatientGeneralTherapy(
+  //     createdData
+  //   );
```
Comment (Osama-you) [inline server/src/database/dummy-data/index.js]: After fixing the naming problem in database shcema, you can uncomment this line.

File: server/src/database/dummy-data/index.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import createFinancialRecords from './financial-records';
+import createAppointments from './appointments';
+
+if (process.env.NODE_ENV !== 'production') {
+  dotenv.config();
+}
+
+const buildData = async () => {
+  const createdData = {};
+  createdData.users = await createUsers(createdData);
```
Comment (Osama-you) [inline server/src/database/dummy-data/index.js]: Why are you passing `createdData` as an argument for `createUsers` function, although it does not take one?
And the same for the rest of the `create` functions.

File: server/src/database/models/patients/schema.sql [hunk 5]
```diff
@@ -0,0 +1,10 @@
+DROP TABLE IF EXISTS "patients" CASCADE;
+
+CREATE TABLE "patients" (
+  "id" SERIAL PRIMARY KEY,
+  "full_name" VARCHAR(50) NOT NULL,
```
Comment (Osama-you) [inline server/src/database/models/patients/schema.sql]: I think the `full_name` is not needed.

File: server/src/database/models/patient-general-therapy/schema.sql (hunk 1 above)
Comment (AlaaShurrab) [reply server/src/database/models/patient-general-therapy/schema.sql]: done

File: server/src/database/dummy-data/patient-general-therapy.js (hunk 2 above)
Comment (AlaaShurrab) [reply server/src/database/dummy-data/patient-general-therapy.js]: done

File: server/src/database/models/patients/schema.sql (hunk 5 above)
Comment (AlaaShurrab) [reply server/src/database/models/patients/schema.sql]: i was following the database scema

**Review by Osama-you — APPROVED** [review]
Great Work

**Pull request merged**